### PR TITLE
Update documentation to better explain uses.

### PR DESCRIPTION
**Pull request by hameerabbasi**

### CODE REVIEW

**Comment by hameerabbasi** [comment]
@rgommers Feel free to merge once this is clear.

File: docs/index.rst [hunk 1]
```diff
@@ -14,6 +14,52 @@
...
+
+What's new in ``uarray``?
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:obj:`uarray` is, to our knowledge, the first backend-system for Python that's generic
+enough to cater to the use-cases of many libraries, while at the same time, being
+library independent.
+
+:obj:`unumpy` is the first approach to leverage :obj:`uarray` in order to build a
+generic backend system for (what we hope) will be the core NumPy API. It will be
```
Comment (rgommers) [inline docs/index.rst]: minor: should be `(what we hope will be)` (bracket place)

File: docs/index.rst [hunk 2]
```diff
@@ -14,6 +14,52 @@
...
+possible to create a backend object, and use that to perform operations. In addition,
+it will be possible to change the used backend via a context manager.
+
+Benefits for End-users
```
Comment (rgommers) [inline docs/index.rst]: `end user` instead of `End-user`

Comment (rgommers) [reply docs/index.rst]: and same below

File: docs/index.rst [hunk 3]
```diff
@@ -14,6 +14,52 @@
...
+End-users can easily take their code written for one backend and use it on another
+backend with a simple switch (using a Python context manager). This can have any number
+of effects, depending on the functionality of the library. For example:
+
+* For Matplotlib, changing styles of plots or producing different windows or image
+  formats.
+* For Tensorly, providing a different computation backend that can be distributed or
+  target the GPU or sparse arrays.
+* For :obj:`unumpy`, it can do a similar thing: provide users with code they already
+  wrote for `unumpy` and easily switch to a different backend.
```
Comment (rgommers) [inline docs/index.rst]: `unumpy` --> `numpy`

**Comment by rgommers** [comment]
Okay fixed the minor comments and merged. Looks good, thanks @hameerabbasi

**Pull request merged**